app/scripts/models/user.js

```diff
@@ -40,14 +43,38 @@ define(function (require, exports, module) {
       // consumers so that they don't have to refetch the account's
       // ephemeral data, e.g. OAuth access tokens.
       this._cachedSignedInAccount = null;
+
+      this.window = options.window || window;
+      this.fetch();
     },
 
     defaults: {
+      // flowId identifies an event flow, which may span across multiple devices.
+      flowId: null,
       // uniqueUserId is a stable identifier for this User on this computer.
       uniqueUserId: null
     },
 
-    resumeTokenFields: ['uniqueUserId'],
+    resumeTokenFields: ['flowId', 'uniqueUserId'],
+
+    resumeTokenSchema: {
+      flowId: vat.hex().len(64),
+      uniqueUserId: vat.uuid()
+    },
+
+    // Hydrate the model. Returns a promise.
+    fetch: function () {
+      var self = this;
+
+      return p()
+        .then(function () {
+          self.populateFromStringifiedResumeToken(self.getSearchParam('resume'));
+          if (! self.has('flowId')) {
+            var flowId = (uuid.v4() + uuid.v4()).replace(/-/g, '');
```

Comment:
Yeah, not sure whether there's a better way to generate hex strings from the client but this seemed like the path of least resistance. I'll move it to a named function to make its intention clearer.